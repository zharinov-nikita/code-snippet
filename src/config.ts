import { TypeConfig } from './types'

export const config: TypeConfig = [
  {
    template: 'ui',
    files: '.cli/ui',
  },
  {
    template: 'page',
    files: '.cli/page',
  },
]
