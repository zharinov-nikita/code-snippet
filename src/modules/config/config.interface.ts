export interface InterfaceConfig<TypeConfig> {
    init: () => void
    get: () => TypeConfig[]
    help: () => void
}
