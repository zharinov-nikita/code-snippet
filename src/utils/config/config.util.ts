import fs from 'fs'
import path from 'path'
import { TypeConfig } from '../../types'

export class UtilConfig {
  public generate(): TypeConfig {
    const snippetsPath = '.code-snippet'
    const snippets = fs.readdirSync(snippetsPath)
    return snippets.map((snippet) => ({ snippet, files: path.join(...[snippetsPath, snippet]) }))
  }
}
