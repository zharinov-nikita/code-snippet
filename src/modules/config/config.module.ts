import fs from 'fs'
import path from 'path'
import { CONSTANTS } from '../../constants'
import { enumPrefixName, enumSnippetName, enumSuffixName } from '../../enums'
import { TypeConfig } from '../../types'
import { InterfaceConfig } from './config.interface'

export class ModuleConfig implements InterfaceConfig<TypeConfig> {
  public readonly rootDirConfig: string
  public readonly isRootDirConfig: boolean

  constructor() {
    this.rootDirConfig = CONSTANTS.ROOT_DIR_CONFIG
    this.isRootDirConfig = fs.existsSync(CONSTANTS.ROOT_DIR_CONFIG)
  }

  private example(index: number): string {
    if (index === 0)
      return `export const SnippetName = [
  '${enumSnippetName.camelCase}',
  '${enumSnippetName.lowerKebabCase}',
  '${enumSnippetName.lowerSnakeCase}',
  '${enumSnippetName.pascalCase}',
  '${enumSnippetName.upperKebabCase}',
  '${enumSnippetName.upperSnakeCase}',
]
    `

    if (index === 1)
      return `export const PrefixNameSnippetName = [
  '${enumPrefixName.camelCase}',
  '${enumPrefixName.lowerKebabCase}',
  '${enumPrefixName.lowerSnakeCase}',
  '${enumPrefixName.pascalCase}',
  '${enumPrefixName.upperKebabCase}',
  '${enumPrefixName.upperSnakeCase}',
      
  '${enumSnippetName.camelCase}',
  '${enumSnippetName.lowerKebabCase}',
  '${enumSnippetName.lowerSnakeCase}',
  '${enumSnippetName.pascalCase}',
  '${enumSnippetName.upperKebabCase}',
  '${enumSnippetName.upperSnakeCase}',
]`

    if (index === 2)
      return `export const SnippetNameSuffixName = [
  '${enumSnippetName.camelCase}',
  '${enumSnippetName.lowerKebabCase}',
  '${enumSnippetName.lowerSnakeCase}',
  '${enumSnippetName.pascalCase}',
  '${enumSnippetName.upperKebabCase}',
  '${enumSnippetName.upperSnakeCase}',
      
  '${enumSuffixName.camelCase}',
  '${enumSuffixName.lowerKebabCase}',
  '${enumSuffixName.lowerSnakeCase}',
  '${enumSuffixName.pascalCase}',
  '${enumSuffixName.upperKebabCase}',
  '${enumSuffixName.upperSnakeCase}',
]`

    if (index === 3)
      return `export const PrefixNameSnippetNameSuffixName = [
  ${enumPrefixName.camelCase}',
  '${enumPrefixName.lowerKebabCase}',
  '${enumPrefixName.lowerSnakeCase}',
  '${enumPrefixName.pascalCase}',
  '${enumPrefixName.upperKebabCase}',
  '${enumPrefixName.upperSnakeCase}',
          
  '${enumSnippetName.camelCase}',
  '${enumSnippetName.lowerKebabCase}',
  '${enumSnippetName.lowerSnakeCase}',
  '${enumSnippetName.pascalCase}',
  '${enumSnippetName.upperKebabCase}',
  '${enumSnippetName.upperSnakeCase}',
        
  '${enumSuffixName.camelCase}',
  '${enumSuffixName.lowerKebabCase}',
  '${enumSuffixName.lowerSnakeCase}',
  '${enumSuffixName.pascalCase}',
  '${enumSuffixName.upperKebabCase}',
  '${enumSuffixName.upperSnakeCase}',
]`

    return `export const SnippetName = [
  '${enumSnippetName.camelCase}',
  '${enumSnippetName.lowerKebabCase}',
  '${enumSnippetName.lowerSnakeCase}',
  '${enumSnippetName.pascalCase}',
  '${enumSnippetName.upperKebabCase}',
  '${enumSnippetName.upperSnakeCase}',
]
    `
  }

  public init(): void {
    const array: number[] = Array.from(Array(4).keys())
    const examplePaths: string[] = array.map((_, index) => `${this.rootDirConfig}/my-snippet-${index + 1}`)

    examplePaths.forEach((pathExample, index) => {
      fs.mkdirSync(path.join(pathExample), { recursive: true })
      fs.writeFileSync(path.join(...[pathExample, 'snippet-name.ts']), this.example(index))
    })
  }

  public get() {
    const snippets = fs.readdirSync(this.rootDirConfig)
    return snippets.map((snippetName) => ({
      snippetName,
      pathToSnippet: path.join(...[this.rootDirConfig, snippetName]),
    }))
  }
}
