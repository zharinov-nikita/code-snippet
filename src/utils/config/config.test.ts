import { describe, expect, test } from '@jest/globals'
import { UtilConfig } from './config.util'

describe('util/config', () => {
  const util = new UtilConfig()

  test('snapshot', () => {
    expect(util).toMatchSnapshot('util')
  })
})
