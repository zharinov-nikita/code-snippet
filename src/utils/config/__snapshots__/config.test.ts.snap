// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`util/config snapshot: util 1`] = `UtilConfig {}`;
